table-api/src/main/java/io/deephaven/api/updateBy/spec/CumProdSpec.java

```diff
@@ -0,0 +1,30 @@
+package io.deephaven.api.updateBy.spec;
+
+import io.deephaven.annotations.SimpleStyle;
+import org.immutables.value.Value.Immutable;
+
+/**
+ * A {@link UpdateBySpec} for performing a Cumulative Product of the specified columns.
+ */
+@Immutable
+@SimpleStyle
+public abstract class CumProdSpec implements UpdateBySpec {
+    public static CumProdSpec of() {
+        return ImmutableCumProdSpec.of();
+    }
+
+    @Override
+    public final boolean applicableTo(Class<?> inputType) {
+        boolean isPrimitiveNumeric = inputType.equals(double.class) || inputType.equals(float.class)
+                || inputType.equals(int.class) || inputType.equals(long.class) || inputType.equals(short.class)
+                || inputType.equals(byte.class);
+        boolean isBoxedNumeric = Number.class.isAssignableFrom(inputType);
+
+        return isPrimitiveNumeric || isBoxedNumeric;
```

Comment:
By separating as two booleans, we don't get lazy evaluation (at least, not without JVM optimization) - so we always call isAssignableFrom. Similar elsewhere.